simpleflow/process/supervisor.py

```diff
@@ -162,15 +162,18 @@ def _handle_sigchld(signum, frame):
             # reliably and determining their state ; we could list children via self._processes
             # but identifying if they're alive is more tricky (???????)
             for child in psutil.Process().children():
-                logger.debug("  child: name=%s pid=%d status=%s" % (child.name(), child.pid, child.status()))
-                if child.status() == psutil.STATUS_ZOMBIE:
-                    logger.debug("  process {} is zombie, will cleanup".format(child.pid))
-                    to_clean = [p for p in self._processes if p.pid == child.pid]
-                    for process in to_clean:
-                        # join process to clean it up
-                        process.join()
-                        # remove the process from self._processes so it will be replaced later
-                        self._processes.remove(process)
+                try:
+                    logger.debug("  child: name=%s pid=%d status=%s" % (child.name(), child.pid, child.status()))
+                    if child.status() == psutil.STATUS_ZOMBIE:
+                        logger.debug("  process {} is zombie, will cleanup".format(child.pid))
+                        to_clean = [p for p in self._processes if p.pid == child.pid]
+                        for process in to_clean:
+                            # join process to clean it up
+                            process.join()
+                            # remove the process from self._processes so it will be replaced later
+                            self._processes.remove(process)
+                except psutil.NoSuchProcess:  # May be untimely deceased
```

Comment:
Hm, obviously we should do something like `name, status = ["unknown", "unknown"]` in the `except` block..